src/main/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/ReadsPreprocessingPipeline.java

```diff
@@ -96,21 +100,39 @@ protected void setupPipeline( Pipeline pipeline ) {
         // Load the Variants and the Reference and join them to reads.
         final VariantsDataflowSource variantsDataflowSource = new VariantsDataflowSource(baseRecalibrationKnownVariants, pipeline);
 
-        Map<String, String> referenceNameToIdTable = RefAPISource.buildReferenceNameToIdTable(pipeline.getOptions(), referenceName);
+        String referenceName = RefAPISource.getReferenceSetID(referenceURL);
+        Map<String, String> referenceNameToIdTable = RefAPISource.getInstance().buildReferenceNameToIdTable(pipeline.getOptions(), referenceName);
         // Use the BQSR_REFERENCE_WINDOW_FUNCTION so that the reference bases required by BQSR for each read are fetched
-        RefAPIMetadata refAPIMetadata = new RefAPIMetadata(referenceName, referenceNameToIdTable, BaseRecalibratorDataflow.BQSR_REFERENCE_WINDOW_FUNCTION);
+        RefAPIMetadata refAPIMetadata = new RefAPIMetadata(referenceName, referenceNameToIdTable, BaseRecalibratorDataflow2.BQSR_REFERENCE_WINDOW_FUNCTION);
 
         final PCollection<KV<GATKRead, ReadContextData>> readsWithContext = AddContextDataToRead.add(markedReads, refAPIMetadata, variantsDataflowSource);
 
         // Apply BQSR.
-        final PCollection<RecalibrationTables> recalibrationReports = readsWithContext.apply(new BaseRecalibratorStub(headerSingleton));
-        final PCollectionView<RecalibrationTables> mergedRecalibrationReport = recalibrationReports.apply(View.<RecalibrationTables>asSingleton());
+        // default arguments are best practice.
+        //BaseRecalibrationArgumentCollection BRAC = new BaseRecalibrationArgumentCollection();
+        //final SAMSequenceDictionary readsDictionary = readsHeader.getSequenceDictionary();
+        //SAMSequenceDictionary refDictionary = RefAPISource.getInstance().getReferenceSequenceDictionary(pipeline.getOptions(), referenceName, readsDictionary);
+        //checkSequenceDictionaries(refDictionary, readsDictionary);
+        //PCollectionView<SAMSequenceDictionary> refDictionaryView = pipeline.apply(Create.of(refDictionary)).setName("refDictionary").apply(View.asSingleton());
+        //final PCollection<BaseRecalOutput> recalibrationReports = readsWithContext.apply(new BaseRecalibratorTransform(headerSingleton, refDictionaryView, BRAC));
+
+        // pretend to apply BQSR
+        final PCollection<BaseRecalOutput> recalibrationReports = readsWithContext.apply(new BaseRecalibratorStub(headerSingleton));
```

Comment:
Apply the actual `BaseRecalibratorTransform` here, and delete the stub. If the `ReadsPreprocessingPipelineIntegrationTest` fails as a result, temporarily disable the test, create a ticket to fix it, and assign it to me.